server/sonar-server/src/main/java/org/sonar/server/computation/ws/ActivityWsAction.java

```diff
@@ -93,7 +105,9 @@ public void define(WebService.NewController controller) {
   public void handle(Request wsRequest, Response wsResponse) throws Exception {
     DbSession dbSession = dbClient.openSession(false);
     try {
-      CeActivityQuery query = readQuery(wsRequest);
+      CeActivityQuery query = buildQuery(dbSession, wsRequest);
+      verifyPermissions(query);
```

Comment:
in general I use checkPermissions as a naming convention. I'd like us to be consistent